### PR TITLE
Migrate to `sbt-http4s-org` plugin

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

**Comment by danicheg** [comment]
@armanbilge @valencik Sorry for that diff 🤷🏻

File: build.sbt [hunk 1]
```diff
@@ -17,6 +19,9 @@ val Scala213 = "2.13.12"
 ThisBuild / crossScalaVersions := Seq(Scala213, "3.3.1")
 ThisBuild / scalaVersion := Scala213
 
+// disable sbt-header plugin until we are not aligned on the license
+ThisBuild / headerCheckAll := Nil
```
Comment (danicheg) [inline build.sbt:23]: https://github.com/http4s/http4s-grpc/issues/98

File: build.sbt [hunk 2]
```diff
@@ -1,13 +1,13 @@
+import explicitdeps.ExplicitDepsPlugin.autoImport.moduleFilterRemoveValue
+
 ThisBuild / tlBaseVersion := "0.0" // your current series x.y
```
Comment (danicheg) [inline build.sbt:3]: I'm willing to set it to `0.1`. Any objections?

**Review by hamnis — APPROVED** [review]

File: codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcCodeGenerator.scala [hunk 3]
```diff
@@ -45,30 +45,27 @@ object Http4sGrpcCodeGenerator extends CodeGenApp {
...
     }.toSeq
-  }
 
-  private def parseParameters(params: String): Either[String, (GeneratorParams)] =
+  private def parseParameters(params: String): Either[String, GeneratorParams] =
     for {
       paramsAndUnparsed <- GeneratorParams.fromStringCollectUnrecognized(params)
       params = paramsAndUnparsed._1
-      unparsed = paramsAndUnparsed._2
+      // unparsed = paramsAndUnparsed._2
```
Comment (valencik) [inline codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcCodeGenerator.scala:53]: Should we just remove this if it's unused?

File: codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcServicePrinter.scala [hunk 4]
```diff
@@ -78,15 +81,17 @@ class Http4sGrpcServicePrinter(service: ServiceDescriptor, di: DescriptorImplici
   }
 
   private[this] def serviceBindingImplementation(method: MethodDescriptor): PrinterEndo = { p =>
-    val serviceCall = s"serviceImpl.${method.name}"
-    val eval = if (method.isServerStreaming) s"$Stream.eval(mkCtx(m))" else "mkCtx(m)"
+    // val serviceCall = s"serviceImpl.${method.name}"
+    // val eval = if (method.isServerStreaming) s"$Stream.eval(mkCtx(m))" else "mkCtx(m)"
```
Comment (valencik) [inline codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcServicePrinter.scala:85]: Similarly. Or perhaps we want to leave that kind of tidying for another PR?

**Review by valencik — APPROVED** [review]
LGTM.
Confirmed scalafix/scalafmt setup are the same as http4s.
I noticed a couple things we could perhaps tidy up, but we could also do so in a followup PR to reduce noise here.

One thing we should do is add the commit where the scalafmt is first run to a `.git-blame-ignore-revs` file like Scala Steward does for us.

So a new `.git-blame-ignore-revs` file with the content:

```
# #97 Introduce sbt-http4s-org plugin
b92d1dbfcc29d3afcca0d2c143528f07b949dcd3
```

File: codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcServicePrinter.scala (hunk 4 above)
Comment (danicheg) [reply codegen/generator/src/main/scala/org/http4s/grpc/generator/Http4sGrpcServicePrinter.scala:85]: I presumed there could be many other refactorings, so I just put this under comments.

**Comment by danicheg** [comment]
Maybe Arman would have something to add. But we already have a quorum of approvers, so merging this.
